src/components/ExperienceTimeline.tsx

```diff
@@ -1,23 +1,77 @@
 import { Box, Text } from "@chakra-ui/react";
-import { ExperienceDetail } from "./Experience";
-import styles from "./ExperienceTimeline.module.css";
+import { Experience } from "../consts";
 
 interface ExperienceTimelineProps {
-  experienceDetails: ExperienceDetail[];
-  updateExperienceDataIndex: (index: number) => void;
+  experienceDetails: Experience[];
+  setExperience: (experience: Experience) => void;
 }
 
+const getPseudoElementStyles = (isHovered: boolean) => {
+  let width = "0.313rem";
+  let height = "0.313rem";
+  let top = "0.625rem";
+  let left = "-0.05rem";
+  let boxShadow =
+    "0 0 0 4px grey, inset 0 2px 0 rgb(0 0 0 / 8%), 0 3px 0 4px rgb(0 0 0 / 5%)";
+  let background = "black";
+
+  if (isHovered) {
```

Comment:
I think we don't need this if block. We can use ternary operators here. To avoid excess cluttering of simple. logic.